share/completions/git.fish

```diff
@@ -622,6 +622,76 @@ function __fish_git_config_keys
     printf '%s\n' (__fish_git help --config)[1..-2] # Last line is a footer; ignore it
 end
 
+# Render the subtree completions, these were infered from the output of `git subtree --help`
+function __fish_render_complete_git_subtree_completion
+    # Define available commands for git subtree
+    function __fish_git_subtree_command
+        echo "add Add a new subtree to the repository"
+        echo "merge Merge changes from a subtree into the repository"
+        echo "split Extract a subtree from the repository"
+        echo "pull Fetch and integrate changes from a remote subtree"
+        echo "push Push changes to a remote subtree"
+    end
+
+    # Define available options for all git subtree commands
+    function __fish_git_subtree_option
+        echo "-q Suppress output"
+        echo "--quiet Suppress output"
+        echo "-d Debug output"
+        echo "--debug Debug output"
+        echo "-P Path to the subtree"
+        echo "--prefix Path to the subtree"
+    end
+
+    # Define options for 'add' and 'merge' commands.
+    # The options remain the same for for pull, split --rejoin, and push --rejoin, according to the help output.
+    function __fish_git_subtree_add_merge_options
+        echo "--squash Squash"
+        echo "-m Message"
+        echo "--message Message"
+    end
+
+    # Define options for 'split' and 'push' commands.
+    # The options remain the same for push --rejoin.
+    function __fish_git_subtree_split_push_options
+        echo "--annotate Annotate the commit"
+        echo "-b Branch to split"
+        echo "--branch Branch to split"
+        echo "--ignore-joins Ignore joins during history reconstruction"
+        echo "--onto Specify the commit ID to start history reconstruction"
+        echo "--rejoin Merge the synthetic history back into the main project"
+    end
+
+    ## git subtree completion when invoking git
+    complete -f -c git -n __fish_git_needs_command -a subtree -d 'Manage git subtrees'
```

Comment:
`subtree` is  in contrib, so not installed by default. Can we only load this one if it exists?
I guess this can be moved to `git-subtree.fish`, then that should happen automatically.